compiler/rustc_metadata/src/creader.rs

```diff
@@ -296,6 +300,83 @@ impl CStore {
         }
     }
 
+    pub fn report_incompatible_target_modifiers(&self, tcx: TyCtxt<'_>, krate: &Crate) {
```

Comment:
I expected this function to just be something like
```
if !tcx.target_modifiers(LOCAL_CRATE) != tcx.target_modifiers(krate) {
    tcx.emit_err(...
}
```
but instead, it's *way* more complicated. Why? What is it doing? It sort of looks like it's trying to come up with a good diagnostic, but I can't even tell if it does the equality check right.

Reply:
Yes, it is for good diagnostic.
I have added such a code for clarity:
```
if mods != *data.target_modifiers() {
    Self::report_target_modifiers_extended(tcx, krate, &mods, data);
}
```
Also, `unsafe_allow_abi_mismatch` need to be taken into account to skip allowed mismatches.
`-Cunsafe-allow-abi-mismatch=regparm,reg-struct-return`.